src/DataStructures/Matrix.hpp

```diff
@@ -25,10 +25,14 @@ class Matrix {
  public:
   /// Create and set each element to value
   Matrix(size_t number_of_rows, size_t number_of_colums,
-         double value = std::numeric_limits<double>::signaling_NaN());
+         double value = std::numeric_limits<double>::signaling_NaN()) noexcept;
+
+  /// Create a matrix from a two-dimensional vector of rows x columns. Requires
+  /// that all rows have the same size.
+  explicit Matrix(const std::vector<std::vector<double>>& rows);
 
   /// Default constructor needed for serialization
-  Matrix() = default;
+  Matrix() noexcept = default;
```

Comment:
Should we actually use the `[]` operator instead of `()` for this, to bring it in line with `DataVector`? Right now the two data structures have a different interface for accessing elements.

Reply:
How? `[]` is always unary.